### PR TITLE
Rename and large update to spec

### PR DESCRIPTION
**Pull request by jawache**
Added a lot more material to the spec.
- Impact Graph
- Durations
- Granularity
- Observations
- Pipeline Processing
- Terminology
- Lots of diagrams

### CODE REVIEW

**Comment by jawache** [comment]
WIP not ready to merge just yet needs some formatting improvement and some pages are still missing but getting this out there to start reviewing and comments.

This solves a lot of problems including HOW  to do Functional Units and HOW to do Grid Emissions but those parts need more fleshing out to explain how this solution answers those points.

File: docs/design/Impact YAML.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+      duration: <duration>
+      <key>: <value>
+
+  # Multiple observations      
+  <component>:
+    model: <imp-identifier>
+    config: 
+      <key>: <value>
+    observations:
+      common:
```
Comment (gnanakeethan) [inline docs/design/Impact YAML.md:68]: I thought we are using config to define any common parameters.

File: docs/design/Impact Model Plugin.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+  * Each model can work with different types of inputs; the spec cannot predict what kinds of inputs will be used by all models, so we need to keep this very open. 
+  * It's an array since we will (in the future) need to deal with GridEMissions (`I`), and that requires input data in a fine grain to make sure we map to `I` at the same granularity. E.g., we might want to output carbon per hour, but the input energy data is in 5min increments so that we can make the grid emissions in the same 5 min increments.
+
+#### Returns
+
+An array of instances of an `ImpactMetric` holds the estimate of emissions.
+
+#### Raises
+
+* `NotAuthorized`
```
Comment (gnanakeethan) [inline docs/design/Impact Model Plugin.md:192]: JSII will not permit specific typed Exceptions. It can only throw generalised Error.

Comment (jawache) [reply docs/design/Impact Model Plugin.md]: Understood.

**Comment by jawache** [comment]
Project agreed to merge

**Pull request merged**